openAPI/core-back-internal.yaml

```diff
@@ -75,7 +75,7 @@ paths:
               \"clientOAuthSessionId\":\"$util.escapeJavaScript($input.params('client-session-id'))\",
               \"language\":\"$util.escapeJavaScript($input.params('language'))\"
               }",
-              "stateMachineArn": "arn:aws:states:${AWS::Region}:${AWS::AccountId}:stateMachine:${JourneyEngineStepFunction.Name}"
+              "stateMachineArn": "arn:aws:states:${AWS::Region}:${AWS::AccountId}:stateMachine:${JourneyEngineStepFunction.Name}:live"
```

Comment:
Do we need to worry about any permissions issues with this change? E.g. I know when we switched to invoking CIMIT lambdas using an alias we had to update our permissions to specify the alias as well.

Reply:
I’ve looked at 3 types of permissions:

1. APIGW to step function with `JourneyEngineStepFunctionApiGateWayIamRole`: No changes needed to this, APIGW can invoke any version of the live alias of the state machine with a response 200

2. Step function to lambdas with LambdaInvokePolicy: Also no changes needed, this is an inbuilt policy and gives permission to invoke a function, alias, or version (https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-policy-template-list.html#lambda-invoke-policy)

3. Test APIGW to lambdas: It looks like the APIGW already has access to `Function.Alias` https://github.com/govuk-one-login/ipv-core-back/blob/eaf5cd912f91798148f77427ef6b5b5c75c7c38e/deploy/template.yaml#L858 - but do you know how I could double check this works for both versions during a canary deployment? Ie what query strings/headers do I need to test permissions from apigw to these lambdas for build-user-identity, initialise-ipv-session, issue-client-access-token etc

Is there anything else I should check? I’ve used https://orch-dev-sre.01.core.dev.stubs.account.gov.uk/ to run through journeys (not sure if I’m doing this bit right?), and it looks like the canaries are splitting traffic, see screenshot from when I upped the ratio from 10:90 to 50:50):

![step-function-core-canaries](https://github.com/user-attachments/assets/fc2ed345-5896-4fc0-9a20-1b02be757944)

Reply:
TBH if you've run through journeys then that's the best test :)